libs/checkpoint/langgraph/checkpoint/serde/encrypted.py

```diff
@@ -0,0 +1,90 @@
+import os
+from typing import Any
+
+from langgraph.checkpoint.serde.base import CipherProtocol, SerializerProtocol
+from langgraph.checkpoint.serde.jsonplus import JsonPlusSerializer
+
+
+class EncryptedSerializer(SerializerProtocol):
+    """Serializer that encrypts and decrypts data using an encryption protocol."""
+
+    def __init__(
+        self, cipher: CipherProtocol, serde: SerializerProtocol = JsonPlusSerializer()
+    ) -> None:
+        self.cipher = cipher
+        self.serde = serde
+
+    def dumps(self, obj: Any) -> bytes:
+        return self.serde.dumps(obj)
+
+    def loads(self, data: bytes) -> Any:
+        return self.serde.loads(data)
+
+    def dumps_typed(self, obj: Any) -> tuple[str, bytes]:
+        """Serialize an object to a tuple (type, bytes) and encrypt the bytes."""
+        # serialize data
+        typ, data = self.serde.dumps_typed(obj)
+        # encrypt data
+        encrypted_data = self.cipher.encrypt(data)
+        # add cipher name to type
+        return f"{typ}+{self.cipher.name}", encrypted_data
+
+    def loads_typed(self, data: tuple[str, bytes]) -> Any:
+        enc_typ, enc_data = data
+        # unencrypted data
+        if "+" not in enc_typ:
+            return self.serde.loads_typed(data)
+        # verify cipher name
+        typ, cipher_name = enc_typ.split("+", 1)
+        if cipher_name != self.cipher.name:
```

Comment:
Is there a world where we should let you provide fallback ciphers for decrypting?